packages/caravan-fees/src/transactionAnalyzer.ts

```diff
@@ -0,0 +1,532 @@
+   * It extracts the recipient address, determines whether it's a change output or not,
+   * and includes the amount in satoshis. The output type is set to "change" if the
+   * output is spendable (typically indicating a change output), and "destination" otherwise.
+   *
+   * @returns {BtcTxOutputTemplate[]} An array of BtcTxOutputTemplate objects representing
+   *          the outputs of the analyzed transaction.
+   */
+  public getOutputTemplates(): BtcTxOutputTemplate[] {
+    return this.outputs.map((output, index) => {
+      return new BtcTxOutputTemplate({
+        address: output.address,
+        type:
+          index === this._changeOutputIndex
+            ? TxOutputType.CHANGE
+            : TxOutputType.DESTINATION,
+        amountSats: output.value as Satoshis,
+      });
+    });
+  }
+  /**
+   * Retrieves the change output of the transaction, if it exists.
+   * @returns {TransactionOutput | null} The change output or null if no change output exists
+   */
+  public getChangeOutput(): TransactionOutput | null {
+    if (this._changeOutputIndex !== undefined) {
+      return this.outputs[this._changeOutputIndex];
+    }
+    return null;
+  }
+
+  // Protected methods
+
+  /**
+   * Deserializes and formats the transaction inputs.
+   * @returns {TransactionInput[]} An array of formatted transaction inputs
+   * @protected
+   */
+  protected deserializeInputs(): TransactionInput[] {
+    return this._originalTx.ins.map((input) => ({
+      txid: input.hash.reverse().toString("hex"),
+      vout: input.index,
+      sequence: input.sequence,
+      scriptSig: input.script.toString("hex"),
+      witness: input.witness.map((w) => w.toString("hex")),
+    }));
+  }
+
+  /**
+   * Deserializes and formats the transaction outputs.
+   * @returns {TransactionOutput[]} An array of formatted transaction outputs
+   * @protected
+   */
+  protected deserializeOutputs(): TransactionOutput[] {
+    return this._originalTx.outs.map((output, index) => ({
+      value: output.value,
+      scriptPubKey: output.script.toString("hex"),
+      address: getOutputAddress(output.script, this._network),
+      isSpendable: index === this._changeOutputIndex,
+    }));
+  }
+
+  /**
+   * Checks if the transaction signals RBF (Replace-By-Fee).
+   * @returns {boolean} True if the transaction signals RBF, false otherwise
+   * @protected
+   */
+  protected isRBFSignaled(): boolean {
+    if (!this._canRBF) {
+      this._canRBF = this._originalTx.ins.some(
+        (input) => input.sequence < 0xfffffffe,
+      );
+    }
+    return this._canRBF;
+  }
+
+  /**
+   * Determines if CPFP (Child-Pays-For-Parent) can be performed on this transaction.
+   * @returns {boolean} True if CPFP can be performed, false otherwise
+   * @protected
+   */
+  protected canPerformCPFP(): boolean {
+    return this.outputs.some(
+      (output) =>
+        !output.isSpendable &&
+        new BigNumber(output.value).isGreaterThan(this._dustThreshold),
```

Comment:
I dont' think this is true. Dusty outputs can be used to perform CPFP

Reply:
this is also the only place in the analyzer where this property is used so could maybe be removed entirely.

Reply:
Thanks for the clarification on this ... it's removed [now ](https://github.com/caravan-bitcoin/caravan/pull/114/commits/47e598fb72e5f5fd182a5ac2453eb4cdd592e75b#diff-b0c4ef170382d75023cd41fe14a8a588936bcc592b31cb86b8abc3178065254fR523)